### PR TITLE
[changelog] Add d2Client as a passable parameter to change capture client config

### PR DESCRIPTION
**Pull request by ZacAttack**
## Add d2Client as a passable parameter to change capture client config

This will enable us to leverage offspring's lifecycle aware to manage the d2clients

## How was this PR tested?
Unit tests

## Does this PR introduce any user-facing changes?

Yes?  but it's a good one I promise!  Basically the underlying d2 client can now be configured and managed by LinkedIn's deployment infra.

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java [hunk 1]
```diff
@@ -69,25 +88,41 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
             newStoreChangelogClientConfig,
             consumer != null ? consumer : new KafkaConsumer<>(newStoreChangelogClientConfig.getConsumerProperties()));
       }
+      newStoreChangelogClientConfig.getInnerClientConfig().setMetricsRepository(metricsRepository);
```
Comment (haoxu07) [inline internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:91]: Just curious why we only set up for `metricsRepo` for After Image use case only? Besides, do we really need this class?

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java [hunk 2]
```diff
@@ -69,25 +88,41 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
...
+      if (viewConfig == null) {
+        throw new VeniceException(
+            "Couldn't retrieve store view information when building change capture client for store " + storeName
+                + " viewName " + viewName);
+      }
+      return viewConfig.getViewClassName();
+    };
+  }
+
+  protected void setViewClassGetter(ViewClassGetter viewClassGetter) {
```
Comment (haoxu07) [inline internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:120]: Nit: When we set `defaultViewClassGetter` it is not default anymore. Besides this set method is only for testing purpose?

File: internal/venice-consumer/src/test/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactoryTest.java [hunk 3]
```diff
@@ -86,10 +132,10 @@ public void testGetChangelogConsumerThrowsException() {
     ChangelogClientConfig globalChangelogClientConfig =
         new ChangelogClientConfig().setConsumerProperties(consumerProperties).setSchemaReader(mockSchemaReader);
     VeniceChangelogConsumerClientFactory veniceChangelogConsumerClientFactory =
-        new VeniceChangelogConsumerClientFactory(globalChangelogClientConfig);
+        new VeniceChangelogConsumerClientFactory(globalChangelogClientConfig, new MetricsRepository());
     D2ControllerClient mockControllerClient = Mockito.mock(D2ControllerClient.class);
 
-    veniceChangelogConsumerClientFactory.setD2ControllerClient(mockControllerClient);
+    // veniceChangelogConsumerClientFactory.setD2ControllerClient(mockControllerClient);
```
Comment (haoxu07) [inline internal/venice-consumer/src/test/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactoryTest.java:138]: If we do not need this mock, then we do not need to mock `mockControllerClient` and no need to prepare `mockControllerClient.getStore(STORE_NAME) ` call below.

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java (hunk 1 above)
Comment (ZacAttack) [reply internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:91]: So it's more in preparation.  We're going to need to add metrics pretty soon, and it seems simpler to get it wired in the short term to the client in offspring.

We right now have zero metrics, and I'm anticipating requests for metrics from clients.  Basically adding it here now saves me an extra PR later.

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java (hunk 2 above)
Comment (ZacAttack) [reply internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:120]: Yeah it's pretty much only for testing.  It was either this or do quite a lot of refactoring in our d2controllerclient.

File: internal/venice-consumer/src/test/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactoryTest.java (hunk 3 above)
Comment (ZacAttack) [reply internal/venice-consumer/src/test/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactoryTest.java:138]: Good point.

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java (hunk 2 above)
Comment (haoxu07) [reply internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:120]: Got it! Maybe just use the name of `viewClassGetter`?

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java (hunk 1 above)
Comment (ZacAttack) [reply internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java]: done.

File: internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java (hunk 2 above)
Comment (ZacAttack) [reply internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java:120]: done

**Review by haoxu07 — APPROVED** [review]
LGTM! Thanks for getting this done!

**Pull request merged**